### PR TITLE
chore: don't verify mock in error tests

### PR DESCRIPTION
**Pull request by enocom**
This test failed in a recent build because the mock server doesn't always receive the expected number of requests. Removing verification solves the flaky test build.

### CODE REVIEW

File: internal/mock/sqladmin.go [hunk 1]
```diff
@@ -53,8 +53,11 @@ func httpClient(requests ...*Request) (*http.Client, string, func() error) {
 		},
 	))
 	// cleanup stops the test server and checks for uncalled requests
```
Comment (kurtisvg) [inline internal/mock/sqladmin.go:55]:
```suggestion
	// cleanup stops the test server and optionally checks for uncalled requests
```

File: internal/cloudsql/instance_test.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ func TestConnectInfoErrors(t *testing.T) {
 	if err != nil {
 		t.Fatalf("%s", err)
 	}
-	defer cleanup()
+	defer cleanup(true)
```
Comment (kurtisvg) [inline internal/cloudsql/instance_test.go:159]: So taking a second look, it seems like we ignore the error in this situations. In that case, do we really need a true/false option or can we just ignore the received value?

We should probably pick one and be consistent - we either pass in a true/false to ignore uncalled endpoints (maybe preferable on the off chance it gets a different error some day), or just all cleanup and ignore the output.

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: internal/cloudsql/instance_test.go (hunk 2 above)
Comment (enocom) [reply internal/cloudsql/instance_test.go]: Sorry -- I was mindlessly typing on the last commit. Yes, let's just cleanup and ignore the error. We'll introduce an argument when we need it, but until this, a simple cleanup without an error check seems best.

File: dialer_test.go [hunk 3]
```diff
@@ -126,9 +126,7 @@ func TestDialWithAdminAPIErrors(t *testing.T) {
 	stop := mock.StartServerProxy(t, inst)
 	defer func() {
 		stop()
-		if err := cleanup(); err != nil {
-			t.Fatalf("%v", err)
-		}
+		cleanup()
```
Comment (kurtisvg) [inline dialer_test.go:129]: nit: Would it be better to use `_ := cleanup()` to indicate we know if returns an error (we just explicitly don't care)

**Review by kurtisvg — APPROVED** [review]

**Pull request merged**